notify.js

```diff
@@ -139,16 +153,30 @@
             'requireInteraction': this.options.requireInteraction,
             'silent': this.options.silent,
             'closeOnClick': this.options.closeOnClick
-        });
+        };
 
-        if (!this.options.requireInteraction && this.options.timeout && !isNaN(this.options.timeout)) {
-            setTimeout(this.close.bind(this), this.options.timeout * 1000);
-        }
+        if (this.options.serviceWorker) {
+            var self = this;
+
+            this.options.serviceWorker.ready.then(function(registration) {
+                registration.showNotification(title, options)
+                .then(function() {
+                    return registration.getNotifications();
+                })
+                .then(function(notifications) {
+                    self.myNotify = notifications[notifications.length - 1]; // TODO: Naive implementation, improve
+                    bindNotificationEvents(self);
```

Comment:
`onnotificationclick` is just an event fired when you click on the notification, similar to `Notification.onclick`. Indeed, it is attached to the serviceWorker object. It is completely unrelated to `show`, `close`, `error`, however. And also, somewhat different to `click`. I think on the ServiceWorker case, the first 3 should be dropped, and we can try to make `click` event somewhat transparent to the user, although I got to think if that is even possible.